polynote-kernel/src/main/scala/polynote/kernel/util/NotebookContext.scala

```diff
@@ -123,7 +123,17 @@ class NotebookContext(implicit concurrent: Concurrent[IO]) {
   def first: Option[CellContext] = Option(_first)
   def last: Option[CellContext] = Option(_last)
 
-  def allResultValues: List[ResultValue] = Option(_last).toList.flatMap(_.visibleValues)
+  /**
+    * @return All the [[ResultValue]]s that are visible from the last cell
+    */
+  def lastScope: List[ResultValue] = Option(_last).toList.flatMap(_.visibleValues)
+
+  /**
+    * @return All the [[ResultValue]]s that are visible from any cell
+    */
+  def allResultValues: List[ResultValue] = Option(_last).toList.flatMap(_.collectBack {
+    case ctx => ctx.resultValues
+  }.flatten)
```

Comment:
Hmm, not sure I understand the difference between these two - what does `visible from any cell` mean? Is it that `lastScope` won't show overrides but `allResultValues` will?